app/controllers/admin/deploy_groups_controller.rb

```diff
@@ -44,8 +44,52 @@ def destroy
     redirect_to action: 'index'
   end
 
+  def deploy_all
+    @stages = matching_stages(deploy_group.environment)
+  end
+
+  def deploy_all_now
+    stages = Stage.find(params.require(:stage_ids))
+    deploys = stages.map do |stage|
+      reference = stage.last_successful_deploy.reference
+      stage = new_stage_with_group(stage) unless matches_group?(stage)
+      deploy_service = DeployService.new(current_user)
+      deploy_service.deploy!(stage, reference: reference)
+    end
+    redirect_to deploys_path(ids: deploys.map(&:id))
+  end
+
   private
 
+  def matching_stages(environment)
+    Project.all.map do |project|
+      project.stages.detect do |stage|
+        matches_group?(stage) && # direct match
+          stage.last_successful_deploy # was deployed
+      end || project.stages.detect do |stage|
+        stage.command.include?("$DEPLOY_GROUPS") && # is dynamic
+          stage.deploy_groups.where(environment: environment).exists? && # is made to go to this environment
+          stage.last_successful_deploy # was deployed
+      end
```

Comment:
was thinking about it ... only used once so far ... so wanted to keep the craziness locale